saltproc/saltproc.py

```diff
@@ -265,108 +500,166 @@ def write_mat_file(self):
-
-        # remove Rb, Sr, Cs, Ba
-        # every 1145 steps = 3435 days
-        if self.current_step % 1145 == 0:
-            discard_id = self.find_iso_indx(['Rb', 'Sr', 'Cs', 'Ba'])
-            self.rem_adens[4, ] = self.remove_iso(discard_id, 1)
-
-        # remove np-237, pu-242
-        # every 1946 steps = 16 years
-        if self.current_step % 1946 == 0:
-            higher_nuc = self.find_iso_indx(['Pu237', 'Pu242'])
-            self.rem_adens[4, ] = self.remove_iso(higher_nuc, 1)
-
-        # refill th232 to keep adens constant
-        # do it every time
-        # if want to do it less often do:
-        # if current_step % time == 0:
-        self.th_adens_db[self.current_step, ] = self.maintain_const(th232_id,
-                                                                    self.th232_adens_0)
-
-        # write the processed material to mat file for next run
-        self.write_mat_file()
+        self.core = self.read_dep()
+
+        # make core values mass by multiplying by volume
+        self.core[self.driver_mat_name] = self.core[self.driver_mat_name] * \
+            self.driver_vol
+        try:
+            self.core[self.blanket_mat_name] = self.core[self.blanket_mat_name] * \
+                self.blanket_vol
+        except:
+            self.core[self.blanket_mat_name] = np.zeros(
+                self.number_of_isotopes)
+        # save pre-processing core mass
+        self.core_mass = {}
+        for key, val in self.core.items():
+            self.core_mass[key] = sum(val)
+
+        # record the depleted composition before reprocessing
+        self.driver_before_db[self.current_step,
+                              :] = self.core[self.driver_mat_name]
+        self.blanket_before_db[self.current_step,
+                               :] = self.core[self.blanket_mat_name]
+
+        # waste / fissile tank db initialization
+        self.waste_tank_db[self.current_step,
+                           :] = self.waste_tank_db[self.current_step-1, :]
+        self.fissile_tank_db[self.current_step,
+                             :] = self.fissile_tank_db[self.current_step-1, :]
+
+        # removal first
+        for group, scheme in self.rep_scheme.items():
+            iso_indx = self.find_iso_indx(scheme['element'])
+            if scheme['to'] == 'waste':
+                # things to dump out
+                self.waste_tank_db[self.current_step, :] += self.remove_iso(iso_indx,
+                                                                            scheme['eff'], scheme['from'])
+                print('REMOVING %f kg of %s FROM %s' %
+                      (self.removed_qty, group, scheme['from']))
+            else:
+                continue
+
+        # get mass lacking in reactor after reprocessing out waste
+        self.core_space = {}
+        for mat, val in self.core.items():
+            self.core_space[mat] = self.core_mass[mat] - sum(val)
+
+        # move things around (fissile to driver)
```

Comment:
make this its own function.

 (and move geometry specific logic out of the main body of saltproc)